other/upgrade_2-1_mysql.sql

```diff
@@ -2609,34 +2609,52 @@ WHERE
 	filename LIKE '%.gif';
 ---#
 
----# Update Settings sets_known
-UPDATE {$db_prefix}settings
-SET value = CONCAT(value, ',alienine')
-WHERE variable = 'smiley_sets_known';
-UPDATE {$db_prefix}settings
-SET value = replace (value, ',aaron', '')
-WHERE variable = 'smiley_sets_known';
-UPDATE {$db_prefix}settings
-SET value = replace (value, ',akyhne', '')
-WHERE variable = 'smiley_sets_known';
-UPDATE {$db_prefix}settings
-SET value = replace (value, ',fugue', '')
-WHERE variable = 'smiley_sets_known';
+---# Cleaning up unused smiley sets and adding the lovely new ones
+---{
+// Start with the prior values...
+$dirs = explode(',', $modSettings['smiley_sets_known']);
+$setnames = explode("\n", $modSettings['smiley_sets_names']);
+$combined = array();
+
+// Confirm they exist in the filesystem; bypass default which is getting removed...
+foreach ($dirs AS $ix => $dir)
+	if (is_dir($modSettings['smileys_dir'] . '/' . $dir . '/') && !empty($setnames[$ix]) && $dir != 'default')
+		$combined[$dir] = $setnames[$ix];
+
+// Add our lovely new smiley sets if not already there...
+if (!array_key_exists('fugue', $combined))
+	$combined['fugue'] = $txt['default_fugue_smileyset_name'];
+if (!array_key_exists('alienine', $combined))
+	$combined['alienine'] = $txt['default_alienine_smileyset_name'];
```

Comment:
Won't this break if $txt is in a different language and these are translated?

Reply:
No, I don't think so.  That is just the user friendly name for the dropdown menu.  Tests fine.  

What *did* concern me about using $txt for the names had more to do with extensibility.  Specifically, does this mean that every new custom smiley set needs to add a $txt value for its name? Where? Or is this specific to the SMF default smileysets?  How would a smileyset creator enable language support?   (Or do I worry too much about such things?)

$txt usage predated this PR & I tried to work with the existing paradigm.  I do think it will work if we consider $txt smileyset names are for SMF sets only.

The only real question I had was about the deletion of the old /default folder.   I really like cleaning up after things...   I know SMF folks don't necessarily like to do so - it's safer to leave it laying around.  So...  I guess my question is whether the deletion step can stay (my preference) or if it should go (more SMF-ish...).  Hope this makes sense...

Reply:
Ok thats fine then.
I would say it should only delete the default files.  Leaving any customized ones alone.

Reply:
OK - then I'll leave it as-is.